src/classical/classical_amg_level.cu

```diff
@@ -548,20 +548,20 @@ void Classical_AMG_Level<TemplateConfig<AMGX_device, t_vecPrec, t_matPrec, t_ind
     if ( spmm_verbose )
     {
         typedef typename Matrix<TConfig_d>::IVector::const_iterator Iterator;
-        typedef thrust::pair<Iterator, Iterator> Result;
+        typedef amgx::thrust::pair<Iterator, Iterator> Result;
```

Comment:
So the guideline will be something like
1) Use `thrust_wrapper` for thrust calls that can allocate temporary workspace
2) In other cases - use `amgx::` for thrust calls/types
?

Reply:
We originally introduced `thrust_wrapper` to handle cases where temporary allocations were made, and now we also use it to reduce synchronisation (current patch), and override stream (future patch).

As I mentioned above, `amgx::thrust` is just basic thrust, necessary to work around issue we had with multiple libraries using thrust (some details are discussed in the release notes https://github.com/NVIDIA/thrust/releases/tag/1.14.0, maybe you saw it before?).

Reply:
Sorry - to clarify, ultimately we probably just want to wrap every thrust call. But it's going to take a lot of testing so I'm going slowly with that.